src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GVMDependenciesNode.cs

```diff
@@ -0,0 +1,178 @@
+
+                            MethodDesc slotDecl;
+                            TypeDesc currentType = potentialOverrideType;
+                            do
+                            {
+                                slotDecl = currentType.ResolveInterfaceMethodToVirtualMethodOnType(genericDefinition);
+                                currentType = currentType.BaseType;
+                            }
+                            while (slotDecl == null && currentType != null);
+
+                            if (slotDecl != null)
+                                CreateDependencyForMethodSlotAndInstantiation(slotDecl, dynamicDependencies, factory);
+                        }
+                    }
+                }
+                else
+                {
+                    // TODO: Ensure GVM Canon Target
+
+                    TypeDesc overrideTypeCanonCur = potentialOverrideType;
+                    TypeDesc methodCanonContainingType = _method.OwningType;
+                    while (overrideTypeCanonCur != null)
+                    {
+                        if (overrideTypeCanonCur.ConvertToCanonForm(CanonicalFormKind.Specific) == methodCanonContainingType.ConvertToCanonForm(CanonicalFormKind.Specific))
+                        {
+                            MethodDesc methodDefInDerivedType = potentialOverrideType.GetMethod(_method.Name, _method.GetTypicalMethodDefinition().Signature);
+                            if(methodDefInDerivedType != null)
+                                CreateDependencyForMethodSlotAndInstantiation(methodDefInDerivedType, dynamicDependencies, factory);
+
+                            MethodDesc slotDecl = MetadataVirtualMethodAlgorithm.FindSlotDefiningMethodForVirtualMethod(_method);
+                            if (slotDecl != null)
+                                CreateDependencyForMethodSlotAndInstantiation(slotDecl.GetMethodDefinition(), dynamicDependencies, factory);
+                        }
+
+                        overrideTypeCanonCur = overrideTypeCanonCur.BaseType;
+                    }
+                }
+            }
+            return dynamicDependencies;
+        }
+
+        private void CreateDependencyForMethodSlotAndInstantiation(MethodDesc methodDef, List<CombinedDependencyListEntry> dynamicDependencies, NodeFactory factory)
+        {
+            Debug.Assert(methodDef != null);
+            Debug.Assert(!methodDef.Signature.IsStatic);
+
+            MethodDesc derivedMethodInstantiation = _method.Context.GetInstantiatedMethod(methodDef, _method.Instantiation);
+
+            // Universal canonical instantiations should be entirely universal canon
+            if (derivedMethodInstantiation.IsCanonicalMethod(CanonicalFormKind.Universal))
+            {
+                derivedMethodInstantiation = derivedMethodInstantiation.GetCanonMethodTarget(CanonicalFormKind.Universal);
+            }
+
+            // TODO: verify for invalid instantiations, like List<void>?
```

Comment:
No... Michal was suggesting we have the constraints validation just do constraints validations, and not check for invalid instantiations. If we need invalid instantiations validation, let it be separate from constraints validation.